Modules/Scripted/Endoscopy/Endoscopy.py

```diff
@@ -834,11 +680,87 @@ def interpolateOrientations(self):
 
                 relativeOrientation = EndoscopyLogic.quaternionToOrientation(quaternion)
 
-                worldOrientation = self.relativeOrientationToWorld(self.resampledCurve, resampledCurvePointIndex, relativeOrientation)
+                worldOrientation = EndoscopyLogic.relativeOrientationToWorld(
+                    self.resampledCurve, resampledCurvePointIndex, relativeOrientation, self.planeNormal,
+                )
 
                 self.resampledCurve.SetNthControlPointOrientation(resampledCurvePointIndex, worldOrientation)
 
-    def getDefaultOrientation(self, curve, curveControlPointIndex):
+    def saveOrientationAtIndex(self, resampledCurvePointIndex, cameraNode):
+        # Compute CameraOrientations dictionary key
+        distanceAlongResampledCurve = EndoscopyLogic.distanceAlongCurveOfNthControlPoint(
+            self.resampledCurve, resampledCurvePointIndex,
+        )
+
+        # Compute CameraOrientations dictionary value
+        cameraPosition = cameraNode.GetPosition()
+        focalPoint = cameraNode.GetFocalPoint()
+        viewUp = cameraNode.GetViewUp()
+        worldMatrix3x3 = EndoscopyLogic.buildCameraMatrix3x3(cameraPosition, focalPoint, viewUp)
+        worldOrientation = EndoscopyLogic.matrix3x3ToOrientation(worldMatrix3x3)
+
+        # Add new dictionary key-value pair
+        cameraOrientations = EndoscopyLogic.getCameraOrientationsFromInputCurve(self.inputCurve)
+        cameraOrientations[distanceAlongResampledCurve] = worldOrientation
+
+        EndoscopyLogic.setInputCurveCameraOrientations(self.inputCurve, cameraOrientations)
```

Comment:
If we change `self.updatingControlPoints` from a `bool` to an `int` then it would be good to wrap this block with
```python
self.updatingControlPoints += 1 # at start of block
self.updatingControlPoints -= 1 # at end of block
```